front/lib/plans/subscription.ts

```diff
@@ -320,23 +318,33 @@ export const pokeUpgradeWorkspaceToPlan = async (
   // This is a way to change the plan limitations but stay on Pro.
   if (isUpgradeToPro) {
     if (
-      activeSubscription &&
-      activeSubscription.sId &&
-      isProPlanCode(activeSubscription.plan.code)
+      !activeSubscription ||
+      !activeSubscription.sId ||
+      !activeSubscription.stripeSubscriptionId
     ) {
-      await Subscription.update(
-        { planId: newPlan.id },
-        {
-          where: {
-            sId: activeSubscription.sId,
-          },
-        }
+      throw new Error(
```

Comment:
It feels weird to throw here 🤔

Reply:
This is the prior behavior so will move forward with it for this refacto but we can def' revisit! I liked the throw because it allowed me to get the error log on prod to see if someone lands into this case (no one should, unless they quick super quickly before the webhook is process after a first subscription).